src/extensions/vb/Microsoft.DotNet.UpgradeAssistant.Steps.VisualBasic/EnableMyDotSupportSubStep.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.VisualBasic
+{
+    /// <summary>
+    /// This analyzer will add &lt;VBRuntime&gt;Embed&lt;/VBRuntime&gt; to the vbproj to resolve compilation errors.
+    /// </summary>
+    /// Per https://github.com/dotnet/runtime/issues/30478#issuecomment-521270193
+    public class EnableMyDotSupportSubStep : UpgradeStep
+    {
+        private readonly VisualBasicProjectUpdaterStep _parent;
+
+        public override string Id => typeof(EnableMyDotSupportSubStep).FullName!;
+
+        public override string Title => "Update vbproj to support \"My.\" namespace";
+
+        public override string Description => "Modifies the vbproj file to address compilation errors.";
+
+        public EnableMyDotSupportSubStep(VisualBasicProjectUpdaterStep vbProjectUpdaterStep, ILogger logger)
+            : base(logger)
+        {
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            ParentStep = _parent = vbProjectUpdaterStep ?? throw new ArgumentNullException(nameof(vbProjectUpdaterStep));
+        }
+
+        protected override Task<bool> IsApplicableImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            // VB updates don't apply until a project is selected
+            if (context?.CurrentProject is null)
+            {
+                return Task.FromResult(false);
+            }
+
+            // applicable to Unit test projects and class libraries
+            return Task.FromResult(context.CurrentProject.OutputType == ProjectOutputType.Library);
+        }
+
+        protected override Task<UpgradeStepInitializeResult> InitializeImplAsync(IUpgradeContext context, CancellationToken token)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            return Task.FromResult(new UpgradeStepInitializeResult(UpgradeStepStatus.Incomplete, $"VB updater \"{Id}\" needs to be applied to {context.CurrentProject!.FileInfo}", BuildBreakRisk.Unknown));
```

Comment:
I don't see any logic here.... why always say incomplete?

Reply:
Agreed. If the `<VBRuntime>` property is already present as expected, this should return complete.